api/v1beta1/machine_types.go

```diff
@@ -230,10 +230,22 @@ type MachineStatus struct {
 	// Conditions defines current service state of the Machine.
 	// +optional
 	Conditions Conditions `json:"conditions,omitempty"`
+
+	// Deletion is the deletion state of the Machine.
+	// +optional
+	Deletion MachineStatusDeletion `json:"deletion,omitempty"`
 }
 
 // ANCHOR_END: MachineStatus
 
+// MachineStatusDeletion is the deletion state of the Machine.
+type MachineStatusDeletion struct {
```

Comment:
Do we need something to indicate that drain cannot yet commence, I would potentially expect a `Drainable` condition that denotes whether lifecycle/pre-drain hooks are preventing the Machine from being deleted yet, do we have that?

Reply:
Right, if the decision has been made, then I'll stay out of the way, carry on as you were 😄